src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -41,6 +41,40 @@ public String getDisplayName() {
     }
 
     /**
+     * Needed if it wants Folders are categorized in Jenkins 2.x.
+     *
+     * TODO: Override when the baseline is upgraded to 2.x
+     * TODO: Replace to {@code NestedProjectsCategory.ID}
+     *
+     * @return A string it represents a ItemCategory identifier.
+     */
+    public String getCategoryId() {
+        return "nestedprojects";
+    }
+
+    /**
+     * Needed if it wants Folders are categorized in Jenkins 2.x.
+     *
+     * TODO: Override when the baseline is upgraded to 2.x
+     *
+     * @return A string with the Item description.
+     */
+    public String getDescription() {
+        return Messages.Folder_Description();
+    }
+
+    /**
+     * Needed if it wants Folder are categorized in Jenkins 2.x.
+     *
+     * TODO: Override when the baseline is upgraded to 2.x
+     *
+     * @return A string it represents a URL pattern to get the Item icon in different sizes.
+     */
+    public String getIconFilePathPattern() {
```

Comment:
:bug: this also should probably be left abstract and implemented in `Folder.DescriptorImpl`, unless you intend to set a default to be used for all `AbstractFolder` subtypes unless overridden.